spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/AvroDataTest.java

```diff
@@ -605,4 +723,172 @@ public void testRowLineage() throws Exception {
             record.copy(Map.of("id", 4L, "data", "d", "_row_id", 1_001L)),
             record.copy(Map.of("id", 5L, "data", "e"))));
   }
+
+  @Test
+  public void testUnshreddedVariant() throws IOException {
```

Comment:
I don't think this is needed because `testTypeSchema(VariantType)` covers the same case, but a bit more thoroughly by including a trailing data field.